From 2d79a0ea2aa41ca5af3969b3e22f6d99b35243b8 Mon Sep 17 00:00:00 2001
From: Pierre-Jean Texier <texier.pj2@gmail.com>
Date: Sun, 13 Nov 2022 13:46:23 +0100
Subject: [PATCH] ARM: WaRP7: Add support to mender

Signed-off-by: Pierre-Jean Texier <pjtexier@koncepto.io>
[Pierre-Jean Texier: refresh patch for dunfell release]
Signed-off-by: Pierre-Jean Texier <texier.pj2@gmail.com>
[Pierre-Jean Texier: refresh patch for kirkstone release]
Signed-off-by: Pierre-Jean Texier <texier.pj2@gmail.com>
---
 configs/warp7_defconfig | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/configs/warp7_defconfig b/configs/warp7_defconfig
index b72332c778..0b6566becc 100644
--- a/configs/warp7_defconfig
+++ b/configs/warp7_defconfig
@@ -8,6 +8,7 @@ CONFIG_DM_GPIO=y
 CONFIG_DEFAULT_DEVICE_TREE="imx7s-warp"
 CONFIG_OPTEE_TZDRAM_SIZE=0x3000000
 CONFIG_TARGET_WARP7=y
+CONFIG_ENV_OFFSET_REDUND=0xE0000
 CONFIG_ARMV7_BOOT_SEC_DEFAULT=y
 # CONFIG_ARMV7_VIRT is not set
 CONFIG_IMX_RDC=y
@@ -43,10 +44,13 @@ CONFIG_CMD_FAT=y
 CONFIG_CMD_FS_GENERIC=y
 CONFIG_OF_CONTROL=y
 CONFIG_ENV_OVERWRITE=y
+CONFIG_SYS_REDUNDAND_ENVIRONMENT=y
 CONFIG_SYS_RELOC_GD_ENV_ADDR=y
 CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG=y
 CONFIG_NET_RANDOM_ETHADDR=y
 CONFIG_BOUNCE_BUFFER=y
+CONFIG_BOOTCOUNT_LIMIT=y
+CONFIG_BOOTCOUNT_ENV=y
 CONFIG_DFU_MMC=y
 CONFIG_SYS_DFU_DATA_BUF_SIZE=0x1000000
 CONFIG_DM_I2C=y
-- 
2.25.1

